GitUI/AutoCompletion/CommitAutoCompleteProvider.cs

```diff
@@ -14,7 +14,10 @@ namespace GitUI.AutoCompletion
 {
     public class CommitAutoCompleteProvider : IAutoCompleteProvider
     {
-        private static readonly Lazy<Dictionary<string, Regex>> _regexes = new Lazy<Dictionary<string, Regex>>(ParseRegexes);
+#pragma warning disable VSTHRD012 // Provide JoinableTaskFactory where allowed
+        private static readonly AsyncLazy<Dictionary<string, Regex>> _regexByExtension = new AsyncLazy<Dictionary<string, Regex>>(ParseRegexesAsync);
+#pragma warning restore VSTHRD012 // Provide JoinableTaskFactory where allowed
```

Comment:
@sharwell could you verify that this diagnostic can be skipped in this scenario? The async operation only awaits disk IO operations so as I see it there's no opportunity for deadlock.

Getting a JTF in this static initialiser is awkward. We'd likely need a new `Initialise` instance method, or convert the ctor to a static factory.